lib/friends/introvert.rb

```diff
@@ -308,32 +295,28 @@ def list_hashtags(from:)
     # The keys of the hash are all of the months (inclusive) between the first
     # and last month in which activities have been recorded.
     #
-    # @param name [String] the name of the friend to use
-    # @return [Hash{String => Fixnum}]
-    # @raise [FriendsError] if 0 or 2+ friends match the given name
-    def graph(name: nil)
-      if name
-        friend = friend_with_name_in(name) # Find the friend by name.
-
-        # Filter out activities that don't include the given friend.
-        acts = @activities.select { |act| act.includes_friend?(friend: friend) }
-      else
-        acts = @activities
-      end
+    # @param with [String] the name of a friend to filter by, or nil for
+    #   unfiltered
+    # @param location_name [String] the name of a location to filter by, or nil
+    #   for unfiltered
+    # @param tagged [String] the name of a hashtag to filter by, or nil for
+    #   unfiltered
+    # @return [Hash{String => Integer}]
+    # @raise [FriendsError] if friend, location or hashtag cannot be found or is ambiguous
+    def graph(with:, location_name:, tagged:)
+      return {} if @activities.empty?
```

Comment:
Things explode further down when calling `@activities.first` and `@activities.last` so it seemed safer to bail here if there's nothing at all to graph.